src/PostgREST/Types.hs

```diff
@@ -199,6 +199,10 @@ type Alias = Text
 type Cast = Text
 type NodeName = Text
 
+-- Rpc query param, only used for GET rpcs
+type RpcQParam = (Text, Text)
+data RpcParam = RpcRW Object | RpcR [RpcQParam]
```

Comment:
This data type `RpcParam ` is not needed and is combining into one two types of information,
- is transaction readonly and the name of the type does not suggest that
- the actual params and they have 2 forms for the same data, i.e. `Object` and `[(key,val)]` which is the same as `Object`

I would say this needs to happen:
- remove this type
- `callProc` maintains the `JSON.Object` param as is, which is calculated in the `App.hs`
- `callProc` gains another Bool (isReadOnly) which is calculated in also in `app`
- optional: the callProc signature is getting ugly, can we invent some kind intermediate record data type to hold together a few of those parameters together?

in short, the section in `app` referring to rpc, does a bit of additional work so that inside `callProc` there is no trace of the form in which the params were sent (get/post)

Reply:
I agree that the RPC params should probably be represented in a single way whether they come in from the post body or the query string.